File: src/main/java/com/deepak/grocery_delivery/service/OrderService.java
package com.deepak.grocery_delivery.service;

import com.deepak.grocery_delivery.dto.order.CreateOrderRequest;
import com.deepak.grocery_delivery.dto.order.OrderResponse;

import java.util.List;

public interface OrderService {

    OrderResponse createOrder(
            String email,
            CreateOrderRequest request
    );

    OrderResponse getOrderById(
            String email,
            Long orderId
    );

    List<OrderResponse> getMyOrders(
            String email
    );

    void cancelOrder(
            String email,
            Long orderId
    );
}

File: frontend/src/services/orderService.js
import api from "./api";

export const createOrder = async (shippingAddress) => {
    const response = await api.post("/api/orders", {
        shippingAddress
    });

    return response.data;
};

export const getMyOrders = async () => {
    const response = await api.get("/api/orders");

    return response.data;
};

export const getOrderById = async (orderId) => {
    const response = await api.get(
        `/api/orders/${orderId}`
    );

    return response.data;
};

export const cancelOrder = async (orderId) => {
    await api.put(
        `/api/orders/${orderId}/cancel`
    );
};